### ISSUE TITLE
Marshalling of sql.Null-Types

### ISSUE THREAD

**Issue by wwwdata** [univedo/api2go#109]
Thanks to @Ramzeus. api2go currently does not serialize the sql.Null Types correctly. They are marshalled as is.

The problem is that the JSON serialization seems to be "wrong" when using api2go in order to serialize the data to JSON. Here is what it might look like:

```json
{  
   "data":[  
      {  
         "accessControlListId":{  
            "Int64":0,
            "Valid":false
         },
         "description":{  
            "String":"",
            "Valid":true
         },
         "displaySchemeId":{  
            "Int64":0,
            "Valid":false
         },
         "id":"379",
         "inactive":false,
         "name":"DISE22",
         "networkId":1,
         "playlistId":{  
            "Int64":0,
            "Valid":false
         },
         "screenDefinitionId":{  
            "Int64":0,
            "Valid":false
         },
         "settingsGroupId":{  
            "Int64":0,
            "Valid":false
         },
         "type":"players"
      }
   ]
}
```
In this case I think this is what would be preferable:
```json
{  
   "data":[  
      {  
         "accessControlListId":null,
         "description":"",
         "displaySchemeId":null,
         "id":"379",
         "inactive":false,
         "name":"DISE22",
         "networkId":1,
         "playlistId":null,
         "screenDefinitionId":null,
         "settingsGroupId":null,
         "type":"players"
      }
   ]
}
```

**Comment by wwwdata** [comment]
I think we already were using `gopkg.in/guregu/null.v2/zero` for this problem. It is a wrapper around the sql.Null Types which also has the interfaces implemented to correctly marshall/unmarshall it. I will write a test with them too and check if this also does not work and fix it. But probably tomorrow.

**Comment by wwwdata** [comment]
Ok, I tried it now with `gopkg.in/guregu/null.v2/zero` and it worked. I will write 2 tests for it that we can merge and I will update the README because this is useful for many people I guess :)

**Comment by wwwdata** [comment]
Hm, I just pushed a commit where marshalling and unmarshaling works. It works but we should think about how we can do the Unmarshalling differently. 

For Marshalling, the Implemented interfaces from the `zero` Library can be used. This just works perfectly. But for Unmarshalling I had to do a lot of type switches for the Zero types. This is not the best solution because it does not use the already implemented `UnmarshalJSON` Methods from that library. 

If we can modify the Unmarshalling to use the "official" go interfaces, this would be compatible with all kinds of fancy custom types that a user wants to, as long as he implements the required 
- https://godoc.org/encoding/json#Marshaler
- https://godoc.org/encoding/json#Unmarshaler

interfaces
